providers/providers.yaml

```diff
@@ -0,0 +1,28 @@
+# This file contains the configuration for supported providers.
+
+# All provider blocks have two kinds of fields:
+# 1) Connector specific, which define properties specific to the go connector itself (e.g. connector_type)
+# 2) Provider specific, which define configuration related to the actual provider API (e.g. provider_base_url)
+# Each field is prefixed accordingly as a convenience and convention.
+
+# This YAML file allows for {{.variables}} (note the period) in the configuration, which can be replaced at runtime
+# by the connector. This syntax (and more) comes from the https://pkg.go.dev/html/template package.
+
+providers:
+  salesforce:
+    connector_type: full
```

Comment:
we need much more granular tracking of supported features than "full" and "basic", https://github.com/amp-labs/server/blob/f38444bef62e823e721e738ddbda5554fd02ad76/shared/common/provider.go#L19 can probably be moved to here.

Reply:
Sure, this is a non-exhaustive list of the configuration, just as a POC. Let's catch up to discuss more